R/exposure_pie_chart.R

```diff
@@ -0,0 +1,103 @@
+#' Create an interactive exposure pie chart in an htmlwidget
+#'
+#' @param .data a exposure pie data frame
+#' @param width width of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param height height of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param ... other options passed on to r2d3::r2d3() (see details)
+#'
+#' @description
+#' The `exposure_pie_chart` function creates an interactive exposure pie chart in an htmlwidget
+#'
+#' @md
+#' @export
+
+exposure_pie_chart <-
+  function(.data, width = NULL, height = NULL, ...) {
+    .data <- export_data_utf8(.data)
+
+    dependencies <-
+      list(
+        system.file("js/pie_exploded.js", package = "r2dii.interactive"),
+        system.file("js/d3-array.v2.min.js", package = "r2dii.interactive"),
+        system.file("js/d3-scale.v3.min.js", package = "r2dii.interactive"),
+        system.file("js/d3-scale-chromatic.v1.min.js", package = "r2dii.interactive"),
+        system.file("js/jquery-3.5.1.js", package = "r2dii.interactive"),
+        system.file("js/text_dropdown_jiggle.js", package = "r2dii.interactive"),
+        system.file("css/2dii_gitbook_style.css", package = "r2dii.interactive"),
+        system.file("css/hide_styles.css", package = "r2dii.interactive")
+        )
+
+    op <- options(r2d3.shadow = FALSE)
+    on.exit(options(op), add = TRUE)
+
+    r2d3::r2d3(
+      data = .data,
+      script = system.file("render_exposure_pie.js", package = "r2dii.interactive"),
+      dependencies = dependencies,
+      d3_version = 4,
+      width = width,
+      height = height,
+      container = "div",
+      ...
+    )
+  }
+
+
+#' Convert raw data into a exposure pie data frame
+#'
+#' @param audit_file audit_file
+#' @param investor_name investor_name
+#' @param portfolio_name portfolio_name
+#' @param twodi_sectors twodi_sectors
+#' @param dataframe_translations dataframe_translations
+#' @param language_select two letter code for language (single string; default = "EN")
+#'
+#' @description
+#' The `as_exposure_pie_data` function converts raw data into a exposure pie data frame
+#'
+#' @import dplyr
+#' @export
+
+as_exposure_pie_data <-
+  function(
+    audit_file,
+    investor_name,
+    portfolio_name,
+    twodi_sectors = c("Power", "Automotive", "Shipping", "Oil&Gas", "Coal", "Steel", "Cement", "Aviation"),
+    dataframe_translations,
+    language_select = "EN"
+  ) {
+    .data <- NULL
+
+    if (missing(dataframe_translations)) {
+      dataframe_translations <- dataframe_translations_default
+    }
+
```

Comment:
Is this empty line here intentional? I see no empty line between the following two calls to `if()`.

Reply:
I felt like the first one is contextually different than the second two, so I intentionally separated them with white space.